app/modals/transaction/transaction-modal.tsx

```diff
@@ -111,29 +113,39 @@ export const TransactionModal: FC<TxModalProps> = ({ balance, address }) => {
     memo: string;
   }
 
-  const createSoftwareWalletTx = useCallback(
-    async (options: CreateTxOptions) => {
-      if (!password || !encryptedMnemonic || !salt) {
-        throw new Error('One of `password`, `encryptedMnemonic` or `salt` is missing');
-      }
-      const { privateKey } = await decryptSoftwareWallet({
-        ciphertextMnemonic: encryptedMnemonic,
-        salt,
-        password,
-      });
-      return makeSTXTokenTransfer({ ...options, senderKey: privateKey });
+  const getSoftwareWalletPrivateKey = useCallback(async () => {
+    if (!password || !encryptedMnemonic || !salt) {
+      throw new Error('One of `password`, `encryptedMnemonic` or `salt` is missing');
+    }
+    const { privateKey } = await decryptSoftwareWallet({
+      ciphertextMnemonic: encryptedMnemonic,
+      salt,
+      password,
+    });
+    return privateKey;
+  }, [encryptedMnemonic, password, salt]);
+
+  const signSoftwareWalletTx = useCallback(
+    async (options: CreateTxOptions & { privateKey: string }) => {
+      return makeSTXTokenTransfer({ ...options, senderKey: options.privateKey });
     },
-    [encryptedMnemonic, password, salt]
+    []
   );
 
-  const createLedgerWalletTx = useCallback(
+  const createUnsignedLedgerTx = useCallback(
     async (options: CreateTxOptions & { publicKey: Buffer }) => {
-      if (!publicKey || !blockstackApp)
-        throw new Error('`publicKey` or `blockstackApp` is not defined');
-      const unsignedTx = await makeUnsignedSTXTokenTransfer({
+      if (!publicKey) throw new Error('`publicKey` is not defined');
+      return makeUnsignedSTXTokenTransfer({
         ...options,
         publicKey: publicKey.toString('hex'),
       });
+    },
+    [publicKey]
+  );
+
+  const signLedgerTransaction = useCallback(
+    async (unsignedTx: StacksTransaction) => {
+      if (!blockstackApp) throw new Error('``blockstackApp` is not defined');
```

Comment:
`hiroApp`??

Reply:
stxApp??

Reply:
STACKS_APP???

Reply:
This code definitely needs to be refactored 😅